lib/Patat/Images/WezTerm.hs

```diff
@@ -0,0 +1,71 @@
+--------------------------------------------------------------------------------
+{-# LANGUAGE TemplateHaskell #-}
+module Patat.Images.WezTerm
+    ( backend
+    ) where
+
+
+--------------------------------------------------------------------------------
+import           Control.Exception           (throwIO)
+import           Control.Monad               (unless, when)
+import           Codec.Picture
+import qualified Data.Aeson                  as A
+import qualified Data.ByteString.Base64      as B64
+import qualified Data.ByteString             as B
+import qualified Data.List                   as L
+import           Patat.Cleanup               (Cleanup)
+import qualified Patat.Images.Internal       as Internal
+import           System.Environment          (lookupEnv)
+
+
+--------------------------------------------------------------------------------
+backend :: Internal.Backend
+backend = Internal.Backend new
+
+
+--------------------------------------------------------------------------------
+data Config = Config deriving (Eq)
+instance A.FromJSON Config where parseJSON _ = return Config
+
+
+--------------------------------------------------------------------------------
+new :: Internal.Config Config -> IO Internal.Handle
+new config = do
+    when (config == Internal.Auto) $ do
+        termProgram <- lookupEnv "TERM_PROGRAM"
+        unless (termProgram == Just "WezTerm") $ throwIO $
+            Internal.BackendNotSupported "TERM_PROGRAM not WezTerm"
+
+    return Internal.Handle {Internal.hDrawImage = drawImage}
+
+
+--------------------------------------------------------------------------------
+drawImage :: FilePath -> IO Cleanup
+drawImage path = do
+    content <- B.readFile path
+    image_dim <- return . getAspectRatio . decodeImage $ content
+    withEscapeSequence $ do
+        putStr "1337;File=inline=1;doNotMoveCursor=1;"
+        putStr image_dim
+        putStr ":"
+        B.putStr (B64.encode content)
+    return mempty
+
+--------------------------------------------------------------------------------
+getAspectRatio :: Either String DynamicImage -> String
```

Comment:
It would be idiomatic to have this function take a `DynamicImage` and do the `Either` check in `drawImage`, e.g.

```
case decodeImage content of
    Left _ -> pure ()
    Right img -> putStr $ getAspectRatio img
```